File: creational/factory/python/concrete_factory.py
from product import Logger
from factory import LoggerFactory
from concrete_product import (
    FileLogger,
    ConsoleLogger,
)


class ConsoleLoggerFactory(LoggerFactory):
    def create_logger() -> Logger:
        return ConsoleLogger()


class FileLoggerFactory(LoggerFactory):
    def create_logger() -> Logger:
        return FileLogger()

File: creational/factory/typescript/concrete-factory.ts
import { ConsoleLogger, FileLogger } from './concrete-product';
import { LoggerFactory } from './factory';
import { Logger } from './product';

export class ConsoleLoggerFactory implements LoggerFactory {
	createLogger(): Logger {
		return new ConsoleLogger();
	}
}

export class FileLoggerFactory implements LoggerFactory {
	createLogger(): FileLogger {
		return new FileLogger();
	}
}
